[CI] SnapshotIT testCreateSnapshot and testSnapshotGetDefaultRepositories failed

Build: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA=java11,ES_RUNTIME_JAVA=java8,nodes=virtual&&linux/61/console

Failure for `testCreateSnapshot`:

```
> Throwable #1: org.elasticsearch.client.ResponseException: method [DELETE], host [http://[::1]:39609], URI [*], status line [HTTP/1.1 400 Bad Request]
   > {"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"Cannot delete indices that are being snapshotted: [[.tasks/SyyqOtuVQyGBbMJc3Yq5PA]]. Try again after snapshot finishes or cancel the currently running snapshot."}],"type":"illegal_argument_exception","reason":"Cannot delete indices that are being snapshotted: [[.tasks/SyyqOtuVQyGBbMJc3Yq5PA]]. Try again after snapshot finishes or cancel the currently running snapshot."},"status":400}
```

Failure for `testSnapshotGetDefaultRepositories`:

```
2> NOTE: leaving temporary files on disk at: /var/lib/jenkins/workspace/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA/java11/ES_RUNTIME_JAVA/java8/nodes/virtual&&linux/client/rest-high-level/build/testrun/integTestRunner/J0/temp/org.elasticsearch.client.SnapshotIT_A5723CC2E76AFBCB-001
   > 	at org.elasticsearch.client.RestClient$SyncResponseListener.get(RestClient.java:687)
   > 	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:218)
   > 	at org.elasticsearch.test.rest.ESRestTestCase.wipeCluster(ESRestTestCase.java:362)
  2> NOTE: test params are: codec=Asserting(Lucene80), sim=Asserting(org.apache.lucene.search.similarities.AssertingSimilarity@6a11b1ca), locale=de-LU, timezone=America/Argentina/Rio_Gallegos
  2> NOTE: Linux 2.6.32-754.6.3.el6.x86_64 amd64/Oracle Corporation 1.8.0_192 (64-bit)/cpus=16,threads=1,free=473172216,total=517996544
   > 	at org.elasticsearch.test.rest.ESRestTestCase.cleanUpCluster(ESRestTestCase.java:193)
  2> NOTE: All tests run in this JVM: [RankEvalIT, TasksClientDocumentationIT, IngestClientDocumentationIT, MachineLearningGetResultsIT, SecurityIT, ClusterClientIT, MigrationDocumentationIT, ReindexIT, TasksIT, PingAndInfoIT, IndicesClientIT, GraphDocumentationIT, MachineLearningIT, BulkRequestWithGlobalParametersIT, StoredScriptsIT, StoredScriptsDocumentationIT, CCRDocumentationIT, MigrationClientDocumentationIT, SnapshotIT]
   > 	at java.lang.Thread.run(Thread.java:748)
   > Caused by: org.elasticsearch.client.ResponseException: method [DELETE], host [http://[::1]:39609], URI [*], status line [HTTP/1.1 400 Bad Request]
   > {"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"Cannot delete indices that are being snapshotted: [[.tasks/SyyqOtuVQyGBbMJc3Yq5PA]]. Try again after snapshot finishes or cancel the currently running snapshot."}],"type":"illegal_argument_exception","reason":"Cannot delete indices that are being snapshotted: [[.tasks/SyyqOtuVQyGBbMJc3Yq5PA]]. Try again after snapshot finishes or cancel the currently running snapshot."},"status":400}
   > 	at org.elasticsearch.client.RestClient$1.completed(RestClient.java:307)
   > 	at org.elasticsearch.client.RestClient$1.completed(RestClient.java:292)
   > 	at org.apache.http.concurrent.BasicFuture.completed(BasicFuture.java:119)
   > 	at org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.responseCompleted(DefaultClientExchangeHandlerImpl.java:177)
   > 	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.processResponse(HttpAsyncRequestExecutor.java:436)
   > 	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.inputReady(HttpAsyncRequestExecutor.java:326)
   > 	at org.apache.http.impl.nio.DefaultNHttpClientConnection.consumeInput(DefaultNHttpClientConnection.java:265)
   > 	at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:81)
   > 	at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:39)
   > 	at org.apache.http.impl.nio.reactor.AbstractIODispatch.inputReady(AbstractIODispatch.java:114)
   > 	at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:162)
   > 	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:337)
   > 	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
   > 	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:276)
   > 	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
   > 	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:588)
   > 	... 1 more
  1> [2018-11-16T01:43:30,339][INFO ][o.e.c.SnapshotIT         ] [testSnapshotGetDefaultRepositories] before test
  1> [2018-11-16T01:43:30,569][INFO ][o.e.c.SnapshotIT         ] [testSnapshotGetDefaultRepositories] There are still tasks running after this test that might break subsequent tests [indices:data/write/bulk, indices:data/write/bulk[s], indices:data/write/index, indices:data/write/reindex].
  1> [2018-11-16T01:43:30,570][INFO ][o.e.c.SnapshotIT         ] [testSnapshotGetDefaultRepositories] after test
FAILURE 0.24s | SnapshotIT.testSnapshotGetDefaultRepositories <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: 
   > Expected: <3>
   >      but: was <2>
   > 	at __randomizedtesting.SeedInfo.seed([A5723CC2E76AFBCB:A5035E9EBAADAB0]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.client.SnapshotIT.testSnapshotGetDefaultRepositories(SnapshotIT.java:94)
   > 	at java.lang.Thread.run(Thread.java:748)
```

Seems like some snapshots got stuck while restoring or took longer.

Could not reproduce locally.

Reproduce with:

```
./gradlew :client:rest-high-level:integTestRunner -Dtests.seed=A5723CC2E76AFBCB -Dtests.class=org.elasticsearch.client.SnapshotIT -Dtests.method="testCreateSnapshot" -Dtests.security.manager=true -Dtests.locale=de-LU -Dtests.timezone=America/Argentina/Rio_Gallegos -Dcompiler.java=11 -Druntime.java=8
```

```
./gradlew :client:rest-high-level:integTestRunner -Dtests.seed=A5723CC2E76AFBCB -Dtests.class=org.elasticsearch.client.SnapshotIT -Dtests.method="testSnapshotGetDefaultRepositories" -Dtests.security.manager=true -Dtests.locale=de-LU -Dtests.timezone=America/Argentina/Rio_Gallegos -Dcompiler.java=11 -Druntime.java=8
```

--- a/client/rest-high-level/src/test/java/org/elasticsearch/client/SnapshotIT.java
+++ b/client/rest-high-level/src/test/java/org/elasticsearch/client/SnapshotIT.java
@@ -144,6 +144,14 @@ public class SnapshotIT extends ESRestHighLevelClientTestCase {
 
         CreateSnapshotResponse response = createTestSnapshot(request);
         assertEquals(waitForCompletion ? RestStatus.OK : RestStatus.ACCEPTED, response.status());
+        if (waitForCompletion == false) {
+            // If we don't wait for the snapshot to complete we have to cancel it to not leak the snapshot task
+            AcknowledgedResponse deleteResponse = execute(
+                new DeleteSnapshotRequest(repository, snapshot),
+                highLevelClient().snapshot()::delete, highLevelClient().snapshot()::deleteAsync
+            );
+            assertTrue(deleteResponse.isAcknowledged());
+        }
     }
 
     public void testGetSnapshots() throws IOException {

